packages/core/src/routes/session/passwordless.ts

```diff
@@ -28,116 +35,40 @@ export default function passwordlessRoutes<T extends AnonymousRouter>(
-      );
-
-      const passcode = await createPasscode(jti, PasscodeType.SignIn, { phone });
-      const { dbEntry } = await sendPasscode(passcode);
-      ctx.log(type, { connectorId: dbEntry.id });
-      ctx.status = 204;
-
-      return next();
-    }
-  );
-
-  router.post(
-    `${signInRoute}/sms/verify-passcode`,
-    koaGuard({ body: object({ phone: string().regex(phoneRegEx), code: string() }) }),
-    async (ctx, next) => {
-      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
-      const { phone, code } = ctx.guard.body;
-      const type = 'SignInSms';
-      ctx.log(type, { phone, code });
-
-      assertThat(
-        await hasUserWithPhone(phone),
-        new RequestError({ code: 'user.phone_not_exists', status: 422 })
-      );
-
-      await verifyPasscode(jti, PasscodeType.SignIn, code, { phone });
-      const { id } = await findUserByPhone(phone);
-      ctx.log(type, { userId: id });
-
-      await updateUserById(id, { lastSignInAt: Date.now() });
-      await assignInteractionResults(ctx, provider, { login: { accountId: id } }, true);
-
-      return next();
-    }
-  );
-
-  router.post(
-    `${signInRoute}/email/send-passcode`,
-    koaGuard({ body: object({ email: string().regex(emailRegEx) }) }),
-    async (ctx, next) => {
-      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
-      const { email } = ctx.guard.body;
-      const type = 'SignInEmailSendPasscode';
-      ctx.log(type, { email });
-
-      assertThat(
-        await hasUserWithEmail(email),
-        new RequestError({ code: 'user.email_not_exists', status: 422 })
-      );
-
-      const passcode = await createPasscode(jti, PasscodeType.SignIn, { email });
-      const { dbEntry } = await sendPasscode(passcode);
-      ctx.log(type, { connectorId: dbEntry.id });
-      ctx.status = 204;
-
-      return next();
-    }
-  );
-
-  router.post(
-    `${signInRoute}/email/verify-passcode`,
-    koaGuard({ body: object({ email: string().regex(emailRegEx), code: string() }) }),
-    async (ctx, next) => {
-      const { jti } = await provider.interactionDetails(ctx.req, ctx.res);
-      const { email, code } = ctx.guard.body;
-      const type = 'SignInEmail';
-      ctx.log(type, { email, code });
-
-      assertThat(
-        await hasUserWithEmail(email),
-        new RequestError({ code: 'user.email_not_exists', status: 422 })
-      );
-
-      await verifyPasscode(jti, PasscodeType.SignIn, code, { email });
-      const { id } = await findUserByEmail(email);
-      ctx.log(type, { userId: id });
-
-      await updateUserById(id, { lastSignInAt: Date.now() });
-      await assignInteractionResults(ctx, provider, { login: { accountId: id } }, true);
-
-      return next();
-    }
-  );
-
-  router.post(
-    `${registerRoute}/sms/send-passcode`,
-    koaGuard({ body: object({ phone: string().regex(phoneRegEx) }) }),
+    '/session/passwordless/:via/send',
```

Comment:
`:via` has only 2 values, can we seperate this into 2 routes? Then we can avoid the type checking for "phone or email", it would be very clean.